curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/6464a6d98c43acecb8a7867b.md

```diff
@@ -208,8 +208,8 @@ const showAnimation = () => {
 const checkUserInput = () => {
   const inputInt = parseInt(numberInput.value);
 
-  if (!numberInput.value || isNaN(inputInt)) {
-    alert("Please provide a decimal number");
+  if (!numberInput.value || isNaN(inputInt) || inputInt < 0) {
+    alert("please provide a decimal number greater than or equal to 0");
     return;
```

Comment:
```suggestion
```
it seems as if there's no utility for this `return` word. can you remove this and check if everything if properly functional or not ? 

Or, if i am missing something, then let me know

Reply:
As the return serves a purpose of stopping the function from continuing when the input is invalid, I am not removing it.

Reply:
oh i see now ! my bad, i didn't see it coming. thank you @DeepaPrasanna